packages/orchestrator/internal/template/server/create_template.go

```diff
@@ -39,30 +40,27 @@ func (s *ServerStore) TemplateCreate(ctx context.Context, templateRequest *templ
 		s.logger.Error("Requesting template creation while server is draining is not possible", logger.WithTemplateID(config.TemplateID))
 		return nil, fmt.Errorf("server is draining")
 	}
-
-	logsWriter := writer.New(
-		s.buildLogger.
-			With(zap.Field{Type: zapcore.StringType, Key: "envID", String: config.TemplateID}).
-			With(zap.Field{Type: zapcore.StringType, Key: "buildID", String: config.BuildID}),
-	)
-
 	template := &build.TemplateConfig{
 		TemplateFiles: storage.NewTemplateFiles(
 			config.TemplateID,
 			config.BuildID,
 			config.KernelVersion,
 			config.FirecrackerVersion,
 		),
-		VCpuCount:       int64(config.VCpuCount),
-		MemoryMB:        int64(config.MemoryMB),
-		StartCmd:        config.StartCommand,
-		ReadyCmd:        config.ReadyCommand,
-		DiskSizeMB:      int64(config.DiskSizeMB),
-		BuildLogsWriter: logsWriter,
-		HugePages:       config.HugePages,
+		VCpuCount:  int64(config.VCpuCount),
+		MemoryMB:   int64(config.MemoryMB),
+		StartCmd:   config.StartCommand,
+		ReadyCmd:   config.ReadyCommand,
+		DiskSizeMB: int64(config.DiskSizeMB),
+		HugePages:  config.HugePages,
 	}
 
-	buildInfo, err := s.buildCache.Create(config.BuildID)
+	logger := s.buildLogger.
+		With(zap.Field{Type: zapcore.StringType, Key: "envID", String: config.TemplateID}).
+		With(zap.Field{Type: zapcore.StringType, Key: "buildID", String: config.BuildID})
+
+	var logs bytes.Buffer
```

Comment:
bytes.Buffer isn't thread safe, writing logs can also cause you will miss some logs when asking for status